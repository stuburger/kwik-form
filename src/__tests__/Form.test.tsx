import * as React from 'react'
// import { Form, Section, Field, Repeat, Gizmo } from '../index'
import { createFormContext } from '../index'
const { Form, Section, Field, Repeat, Gizmo } = createFormContext()
import renderer from 'react-test-renderer'
const initialValue = {
  firstName: 'John',
  lastName: 'Woodcock',
  age: 30,
  contact: {
    tel: '983838333'
  },
  friends: ['stuart', 'oumar']
  // hobbies: [{ type: 'art', name: 'print making' }, undefined, undefined]
}

const defaultValue = {
  firstName: '',
  lastName: '',
  age: 0,
  contact: {
    tel: '',
    email: 'asdf@testing.com'
  }
  // friends: [],
  // hobbies: []
}

function TextInput(props: any) {
  return (
    <div>
      <label>{props.label}</label>
      <input {...props.input} />
    </div>
  )
}

describe('Form', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Form initialValue={initialValue} defaultValue={defaultValue}>
          <Field name="firstName" label="First Name" component={TextInput} />
          <Section name="contact">
            <Field name="tel" label="First Name" component={TextInput} />
            <Field name="email" label="Email Address" component={TextInput} />
          </Section>
          <Repeat<{ type: string; name: string }>
            name="hobbies"
            fallback={[{ type: 'art', name: 'print making' }, undefined, undefined] as any}
          >
            {hobbies => {
              return hobbies.map((value, i) => {
                return (
                  <Section name={i} key={i} fallback={{ type: 'new', name: 'new' }}>
                    <Field name="type" label="Type of hobby" component={TextInput} />
                    <Field name="name" label="Name of hobby" component={TextInput} />
                  </Section>
                )
              })
            }}
          </Repeat>
          <Gizmo
            render={props => {
              return <pre>{JSON.stringify(props, null, 2)}</pre>
            }}
          />
        </Form>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
