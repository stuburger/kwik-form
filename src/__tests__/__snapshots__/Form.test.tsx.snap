// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
Array [
  <div>
    <label>
      First Name
    </label>
    <input
      name="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="John"
    />
  </div>,
  <div>
    <label>
      First Name
    </label>
    <input
      name="tel"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="983838333"
    />
  </div>,
  <div>
    <label>
      Email Address
    </label>
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="asdf@testing.com"
    />
  </div>,
  <div>
    <label>
      Type of hobby
    </label>
    <input
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="art"
    />
  </div>,
  <div>
    <label>
      Name of hobby
    </label>
    <input
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="print making"
    />
  </div>,
  <div>
    <label>
      Type of hobby
    </label>
    <input
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="new"
    />
  </div>,
  <div>
    <label>
      Name of hobby
    </label>
    <input
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="new"
    />
  </div>,
  <div>
    <label>
      Type of hobby
    </label>
    <input
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="new"
    />
  </div>,
  <div>
    <label>
      Name of hobby
    </label>
    <input
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value="new"
    />
  </div>,
  <pre>
    {
  "activeField": null,
  "visited": {},
  "submitCount": 0,
  "errors": {},
  "touched": {},
  "formValue": {
    "firstName": "John",
    "lastName": "Woodcock",
    "age": 30,
    "contact": {
      "tel": "983838333",
      "email": "asdf@testing.com"
    },
    "friends": [
      "stuart",
      "oumar"
    ]
  },
  "defaultValue": {
    "firstName": "",
    "lastName": "",
    "age": 0,
    "contact": {
      "tel": "",
      "email": "asdf@testing.com"
    }
  },
  "initialValue": {
    "firstName": "John",
    "lastName": "Woodcock",
    "age": 30,
    "contact": {
      "tel": "983838333"
    },
    "friends": [
      "stuart",
      "oumar"
    ]
  },
  "formIsValid": true,
  "formIsTouched": false,
  "formIsDirty": false
}
  </pre>,
]
`;
